File: common_library/src/main/java/com/phone/common_library/manager/GetJsonDataManager.java
package com.phone.common_library.manager;


import android.content.Context;
import android.content.res.AssetManager;

import com.phone.common_library.BaseApplication;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

/**
 * author    : xxxxxxxxxxx
 * e-mail    : xxxxxxxxxxx@qq.com
 * date      : 2020/4/11 10:40
 * introduce :
 */


public class GetJsonDataManager {

    public String getJson(String fileName) {

        StringBuilder stringBuilder = new StringBuilder();
        try {
            AssetManager assetManager = BaseApplication.getInstance().getAssets();
            BufferedReader bf = new BufferedReader(new InputStreamReader(
                    assetManager.open(fileName)));
            String line;
            while ((line = bf.readLine()) != null) {
                stringBuilder.append(line);
            }
        } catch (IOException e) {
            e.printStackTrace();
        }
        return stringBuilder.toString();
    }
}

File: library_common/src/main/java/com/phone/library_common/manager/GetJsonDataManager.kt
package com.phone.library_common.manager

import com.phone.library_common.BaseApplication
import java.io.BufferedReader
import java.io.IOException
import java.io.InputStreamReader

/**
 * 不會造成App崩潰的異常（被try(){}catch拋出來的異常）管理類
 */
class GetJsonDataManager {

    fun getJson(fileName: String?): String {
        val stringBuilder = StringBuilder()
        try {
            val assetManager = BaseApplication.get().assets
            val bf = BufferedReader(
                InputStreamReader(
                    assetManager.open(fileName!!)
                )
            )
            var line: String?
            while (bf.readLine().also { line = it } != null) {
                stringBuilder.append(line)
            }
        } catch (e: IOException) {
            e.printStackTrace()
        }
        return stringBuilder.toString()
    }
}